components/webui/client/src/pages/SearchPage/SearchControls/Presto/SqlInterfaceButton/index.tsx

```diff
@@ -11,13 +23,58 @@ import styles from "./index.module.css";
  * @return
  */
 const SqlInterfaceButton = () => {
+    const {token} = theme.useToken();
     const sqlInterface = usePrestoSearchState((state) => state.sqlInterface);
+    const searchUiState = useSearchStore((state) => state.searchUiState);
+    const disabled = searchUiState === SEARCH_UI_STATE.QUERY_ID_PENDING ||
+        searchUiState === SEARCH_UI_STATE.QUERYING;
+
+    const handleChange = (value: PRESTO_SQL_INTERFACE) => {
+        if (value === sqlInterface) {
+            return;
+        }
+
+        if (value === PRESTO_SQL_INTERFACE.GUIDED) {
+            handleSwitchToGuided();
+
+            return;
+        }
+
+        handleSwitchToFreeform();
+    };
```

Comment:
It might be safer to `useCallback`, even though I cannot find a case of outdated `sqlInterface`?

```suggestion
    const handleChange = useCallback((value: PRESTO_SQL_INTERFACE) => {
        if (value === sqlInterface) {
            return;
        }

        if (value === PRESTO_SQL_INTERFACE.GUIDED) {
            handleSwitchToGuided();

            return;
        }

        handleSwitchToFreeform();
    }, [sqlInterface]);

```